### PR TITLE
Feature/graph ql payment

### PR DESCRIPTION
**Pull request by swift-student**
## What's here (& why)

- Method for fetching payments by property id
- Fixed date decoding
- Clean up mutations and queries

### CODE REVIEW

File: EcoSoapBank/Model Controller/PaymentController.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Foundation
 import Combine
 
 protocol PaymentDataProvider {
-    func fetchPayments(forPropertyId propertyId: String,
+    func fetchPayments(forPropertyID propertyId: String,
```
Comment (DeVitoC) [inline EcoSoapBank/Model Controller/PaymentController.swift:13]: Why this change?

File: EcoSoapBank/Model Controller/PaymentController.swift [hunk 2]
```diff
@@ -32,7 +32,7 @@ class PaymentController {
     func fetchAllPayments(forPropertyId propertyId: String) -> Future<[Payment], Error> {
 
         Future { promise in
-            self.dataProvider.fetchPayments(forPropertyId: propertyId) { [weak self] result in
+            self.dataProvider.fetchPayments(forPropertyID: propertyId) { [weak self] result in
```
Comment (DeVitoC) [inline EcoSoapBank/Model Controller/PaymentController.swift:35]: And why this one?

File: EcoSoapBank/Model/Payment.swift [hunk 3]
```diff
@@ -12,8 +12,8 @@ struct Payment: Decodable {
     let id: String
     let invoiceCode: String?
     let invoice: String?
-    let amountPaid: String
-    let amountDue: String
+    let amountPaid: Int
+    let amountDue: Int
     let date: Date
```
Comment (DeVitoC) [inline EcoSoapBank/Model/Payment.swift:17]: Yeah, I realized this when setting up my mock data. Hopefully both of us changing this independently won't cause a conflict.

File: EcoSoapBank/Helpers/DateCoding.swift [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    static let graphQL = JSONDecoder.DateDecodingStrategy.custom { decoder -> Date in
+        let container = try decoder.singleValueContainer()
+        let dateString = try container.decode(String.self).replacingOccurrences(of: "−", with: "-")
+        print("Decoding date: \(dateString)")
+        guard let date = ISO8601DateFormatter.full.date(from: dateString) ??
+            ISO8601DateFormatter.short.date(from: dateString) else {
+                throw NSError(domain: "Unable to parse date", code: 0)
+        }
+        return date
+    }
```
Comment (jonbash) [inline EcoSoapBank/Helpers/DateCoding.swift:21]: Nice, it's good to know this exists.

Comment (jonbash) [reply EcoSoapBank/Helpers/DateCoding.swift:21]: I wonder if we should eventually make a custom error there?

File: EcoSoapBank/Model Controller/PaymentController.swift (hunk 2 above)
Comment (swift-student) [reply EcoSoapBank/Model Controller/PaymentController.swift:35]: I did this for consistency. Since the GraphQLController will have this method sitting next to all of the other data provider methods, I wanted it to match the others which use the format `somethingID`.
